### PR TITLE
Support 9.0 and 9.1 and add travis. 

### PR DESCRIPTION
**Pull request by mrow4a**

### CODE REVIEW

File: js/paperhive.js [hunk 1]
```diff
@@ -18,363 +18,367 @@
...
+                // Get the file data from PaperHive API
+                var dir = fileList.getCurrentDirectory();
+                var $saveNot = OC.Notification.showHtml(t('files_paperhive', 'Saving...'));
+                OCA.Files_PaperHive.getPaperHiveBook(
+                    dir,
+                    bookID,
+                    function (paperHiveData) {
+                        var filename = paperHiveData.filename+paperHiveData.extension;
+
+                        fileList.filesClient.getFileInfo(
```
Comment (mrow4a) [inline js/paperhive.js]: Can I use filesClient in here to do PROPFIND for this file? I want to do this in order to add file into the list (can I get all the properties from PUT to serever?

File: controller/paperhivecontroller.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 /**
  * @author Piotr Mrowczynski <piotr.mrowczynski@yahoo.com>
  *
- * @copyright Copyright (c) 2017, ownCloud, Inc.
+ * @copyright Copyright (c) 2017, ownCloud.
```
Comment (DeepDiver1975) [inline controller/paperhivecontroller.php]: the copyright is with you @mrow4a

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+before_install:
+  - php --info
+  # Set up DB
+  - if [[ "$DB" == 'pgsql' ]]; then createuser -U travis -s oc_autotest; fi
+  - if [[ "$DB" == 'mysql' ]]; then mysql -u root -e 'CREATE DATABASE oc_autotest;'; fi
+  - if [[ "$DB" == 'mysql' ]]; then mysql -u root -e "CREATE USER 'oc_autotest'@'localhost' IDENTIFIED BY '';"; fi
+  - if [[ "$DB" == 'mysql' ]]; then mysql -u root -e "GRANT ALL ON oc_autotest.* TO 'oc_autotest'@'localhost';"; fi
+
+  - sh -c "if [ '$TRAVIS_PHP_VERSION' = '5.4' ]; then curl -s -o $HOME/.phpenv/versions/5.4/bin/phpunit https://phar.phpunit.de/phpunit-4.8.9.phar; fi;"
```
Comment (DeepDiver1975) [inline .travis.yml]: no 5.4

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  - sh -c "if [ '$TRAVIS_PHP_VERSION' = '5.4' ]; then chmod +x $HOME/.phpenv/versions/5.4/bin/phpunit; fi;"
+  - composer self-update
+  - composer install
+  - cd ..
+  - sh -c "if [ '$CLOUD' = 'owncloud' ]; then git clone https://github.com/owncloud/core.git --recursive --depth 1 -b $CORE_BRANCH core; fi;"
+  - sh -c "if [ '$CLOUD' = 'owncloud' ] && [ $CORE_BRANCH = 'master' ]; then cd core && make; fi;"
+  - sh -c "if ([ '$CLOUD' = 'owncloud' ] && [ $CORE_BRANCH != 'master' ]) || [ '$CLOUD' = 'nextcloud' ]; then cd core && git submodule update --init; fi;"
+  - mv music core/apps/
```
Comment (DeepDiver1975) [inline .travis.yml]: music?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    - OC_PASS=ampache
+  matrix:
+    - DB=mysql;CLOUD=owncloud
+
+matrix:
+  include:
+    #- php: 5.6
+      #env: DB=mysql CLOUD=owncloud CORE_BRANCH=stable8.1
+    - php: 7
+      env: DB=mysql CLOUD=owncloud CORE_BRANCH=stable8.2
```
Comment (DeepDiver1975) [inline .travis.yml]: according to info.xml 8.2 is not supported

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
+language: php
+sudo: false
+php:
+  - 5.6
+  - 7
+
+env:
+  global:
+    - CORE_BRANCH=master
+    - OC_PASS=ampache
```
Comment (DeepDiver1975) [inline .travis.yml]: needed?

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  - sh -c "if [ '$CLOUD' = 'owncloud' ]; then git clone https://github.com/owncloud/core.git --recursive --depth 1 -b $CORE_BRANCH core; fi;"
+  - sh -c "if [ '$CLOUD' = 'owncloud' ] && [ $CORE_BRANCH = 'master' ]; then cd core && make; fi;"
+  - sh -c "if ([ '$CLOUD' = 'owncloud' ] && [ $CORE_BRANCH != 'master' ]) || [ '$CLOUD' = 'nextcloud' ]; then cd core && git submodule update --init; fi;"
+  - mv music core/apps/
+
+before_script:
+  # setup ownCloud
+  - php -f core/occ maintenance:install --database-name oc_autotest --database-user oc_autotest --admin-user admin --admin-pass admin --database $DB --database-pass=''
+  - php -f core/occ app:enable files_paperhive
+  - php -f core/occ user:add ampache --password-from-env
```
Comment (DeepDiver1975) [inline .travis.yml]: user ampache?

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+script:
+  - phpunit --coverage-clover clover-unit.xml --configuration tests/php/unit/phpunit.xml tests/php/unit
+  - phpunit --coverage-clover clover-integration.xml --configuration tests/php/integration/phpunit.xml tests/php/integration
+  - cd tests && ../vendor/bin/behat
```
Comment (DeepDiver1975) [inline .travis.yml]: behat?

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  - php --info
+  # Set up DB
+  - if [[ "$DB" == 'pgsql' ]]; then createuser -U travis -s oc_autotest; fi
+  - if [[ "$DB" == 'mysql' ]]; then mysql -u root -e 'CREATE DATABASE oc_autotest;'; fi
+  - if [[ "$DB" == 'mysql' ]]; then mysql -u root -e "CREATE USER 'oc_autotest'@'localhost' IDENTIFIED BY '';"; fi
+  - if [[ "$DB" == 'mysql' ]]; then mysql -u root -e "GRANT ALL ON oc_autotest.* TO 'oc_autotest'@'localhost';"; fi
+
+  - sh -c "if [ '$TRAVIS_PHP_VERSION' = '5.4' ]; then curl -s -o $HOME/.phpenv/versions/5.4/bin/phpunit https://phar.phpunit.de/phpunit-4.8.9.phar; fi;"
+  - sh -c "if [ '$TRAVIS_PHP_VERSION' = '5.4' ]; then chmod +x $HOME/.phpenv/versions/5.4/bin/phpunit; fi;"
+  - composer self-update
```
Comment (DeepDiver1975) [inline .travis.yml]: composer - really? where?

File: .travis.yml (hunk 3 above)
Comment (mrow4a) [inline .travis.yml]: What is the correct one though?

File: tests/controller/paperhivecontrollertest.php [hunk 10]
```diff
@@ -2,7 +2,7 @@
 /**
  * @author Piotr Mrowczynski <piotr.mrowczynski@yahoo.com>
  *
- * @copyright Copyright (c) 2017, ownCloud, Inc.
+ * @copyright Copyright (c) 2017, ownCloud.
```
Comment (DeepDiver1975) [inline tests/controller/paperhivecontrollertest.php]: copy right

File: .travis.yml [hunk 11]
```diff
@@ -0,0 +1,37 @@
+language: php
+php:
+  - 5.6
+  - 7
```
Comment (DeepDiver1975) [inline .travis.yml]: please add 7.1 for master
5.4 for 8.2, 9.0 and 9.1

**Pull request merged**